# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: servo/ports/geckolib/glue.rs
# Commit: 3a8c6048c5d0
# Full Hash: 3a8c6048c5d0874dd3d32c06e7923625b9383e8d
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-05-09 15:34:57
# Regressor Bug: 1545177
# File Overlap Count: 2
# Description:
#   Bug 1545177 - If descriptors of a FontFace are modified after creation, update the associated font entry so that face selection will respect the new values, and mark font sets as dirty. r=heycam
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27980
# ==============================================================================

diff -r c24c083425d0 -r 3a8c6048c5d0 servo/ports/geckolib/glue.rs
--- a/servo/ports/geckolib/glue.rs	Thu May 09 06:41:13 2019 +0000
+++ b/servo/ports/geckolib/glue.rs	Thu May 09 09:43:42 2019 +0000
@@ -2875,6 +2875,7 @@
     desc: nsCSSFontDesc,
     value: *const nsACString,
     data: *mut URLExtraData,
+    out_changed: *mut bool,
 ) -> bool {
     let value = value.as_ref().unwrap().as_str_unchecked();
     let mut input = ParserInput::new(&value);
@@ -2900,7 +2901,9 @@
                     $(
                         nsCSSFontDesc::$v_enum_name => {
                             if let Ok(value) = parser.parse_entirely(|i| Parse::parse(&context, i)) {
-                                rule.$field = Some(value);
+                                let result = Some(value);
+                                *out_changed = result != rule.$field;
+                                rule.$field = result;
                                 true
                             } else {
                                 false
